### ISSUE TITLE
Support for New REST Api 

### ISSUE THREAD

**Issue by timonf** [academe/Omnipay-Wirecard#14]
Hey there,

there seems to be a new API by Wirecard: https://document-center.wirecard.com/display/PTD

Can this be implemented as a Gateway, too?

**Comment by jonakieling** [comment]
Hi @timonf ,

we're looking into this too, since the current API will be disabled in the future. Have you found a different solution for omnipay or a different approach altogether? I'm guessing because the issue is now some time old.

kind regards
jona

**Comment by innomedia** [comment]
Did anyone find a way to support the new API? We are in need of a solution too.

**Comment by guilty80100** [comment]
Have you found any solutions for the new API with omnipay. I've been searching all over github and I only found this : https://github.com/wirecard/wpp-integration-demo-php

I'm very surprised there is no wirecard WPP gateway available knowing that WCP will no longer work in October.

**Comment by innomedia** [comment]
We've written our own implementation that supports purchase/refund for CreditCard, EPS, Sofort and giropay though it's a quick and very dirty one. you can take a look https://github.com/innomedia/omnipay_wirecard be warned though I just ripped it out of the one Project where we've implemented so it will need some adapting because it depends on some project specific data especially inside the Purchase Request/Response and or Refund Request/Response classes (also mentioned in the README)

**Comment by judgej** [comment]
They have not made it easy to access the specs. When I last worked on this some time go, I did not even need a login, and now many of the specs are behind logins. I'll take a look and see how easy it will be to upgrade for the REST API.

**Comment by chromos33** [comment]
Don't know if you know but at least for me the specs/documentation are 1 google away https://doc.wirecard.com/

**Comment by judgej** [comment]
I was looking for the OpenAPI specs, as we can build code very quickly from that, but that was only available through page behind the developer portal login. However, Wirecard offer a lot of different services, so that may have been an API spec for a different service altogether.

Registering for the developer portal was a nightmare - strict password requirements, but passord managers are all disabled through JavaScript on the registration form. That doesn't fill me with confidence at all. There security "features" have resuled in me compromising on security. Not good.

Anyway, thanks fot the link :-)

**Comment by innomedia** [comment]
Yeah! Wirecard is a nightmare. Took us 8 weeks to get working credentials. Service especially in munich, germany is hell. The austrian guys did a good job an helped out a lot. Seems like what the press tells is true for technical things too...

**Comment by judgej** [comment]
There is an OpenAPI spec here https://github.com/wirecard/docs-oas/blob/master/payment-gateway.yml

I'm hoping that can generate most of the models for us at least.

I have not got the time to look at this until the weekend at the earliest, but this could help move things much faster.

![image](https://user-images.githubusercontent.com/395934/86620287-fb09dc00-bfb3-11ea-8f5e-e3952903e6f1.png)

SDK here also: https://github.com/wirecard/paymentSDK-php

**Comment by guilty80100** [comment]
Any news ? 
I know you are doing this on your free time and without being paid for it and I'm very gratefull for it. Do you think, you'll be supporting the new api when the old one is expired or will there be a period without support ? 
I need to prepare my website in case wirecard is no longer an option in October.
